ordering/__init__.py

```diff
@@ -61,6 +61,16 @@ def compare(self, left_item: T, right_item: T) -> bool:
 
         return self._labels[left_item] < self._labels[right_item]
 
+    def reverse(self):
+        self._successors, self._predecessors = self._predecessors, self._successors
+        self._successors[self._start] = self._successors.pop(self._end)
+        self._predecessors[self._end] = self._predecessors.pop(self._start)
+
+        labels = self._labels
+        as_list = list(self)
+        for a, b in zip(as_list[:len(self) // 2], reversed(as_list)):
+            labels[a], labels[b] = labels[b], labels[a]
```

Comment:
I did have it that way originally, but I feel that swapping the labels is better because it is agnostic of the actual values of the labels, and avoids any cost of creating new labels.